packages/cli/internal/pkg/cli/workflow/manager.go

```diff
@@ -436,6 +475,31 @@ func (m *Manager) setWorkflowParameters() {
 	m.workflowParams["workflowInputs"] = filepath.Base(m.attachments[0])
 }
 
+func (m *Manager) setWorkflowEngineParameters() {
+	if m.err != nil {
+		return
+	}
+	m.workflowEngineParams = make(map[string]string)
+	if m.optionFileUrl == "" {
+		return
+	}
+	if m.options != nil {
+		if m.workflowEngine == "nextflow" || m.workflowEngine == "miniwdl" {
+			log.Error().Msgf("optionFile flag cannot be used with head node engines")
+			return
+		}
+		m.workflowEngineParams = m.options
+	}
+
+	namePattern := fmt.Sprintf("%s_*", filepath.Base(m.optionFileUrl))
+	optionFileName, err := writeToTmp(namePattern, m.optionFileUrl)
+	if err != nil {
+		m.err = err
+		return
+	}
+	m.attachments = append(m.attachments, optionFileName)
```

Comment:
How does this work? We're both passing the option file as an attachment and as a map of options. It feels redundant - shouldn't only one would be needed?

Reply:
We don't need to save it as an attachment